pkg/datapath/iptables/iptables.go

```diff
@@ -181,9 +235,15 @@ var ciliumChains = []customChain{
 
 // RemoveRules removes iptables rules installed by Cilium.
 func RemoveRules() {
+	// Set of tables that has had iptables rules in any Cilium version
 	tables := []string{"nat", "mangle", "raw", "filter"}
 	for _, t := range tables {
-		removeCiliumRules(t)
+		removeCiliumRules(t, "iptables")
+	}
+	// Set of tables that has had ip6tables rules in any Cilium version
+	tables6 := []string{"mangle", "raw"}
```

Comment:
Is this a bug? Why weren't we removing IPv6 rules before?

This commit is huge, _and_ it's iptables :fearful:  Any chance to split it up?

(I'm giving up on the iptables-specific part of the review for now, I can take a fresh look later).

Reply:
We weren't adding IPv6 rules before.

Suggestions for how/what to split off are welcome :-)

Reply:
Looking at "bpf: Use iptables TPROXY for proxy redirects ", there's a lot of stuff - total about 2k lines of changes total. I'm guessing that at some point the easiest way to test was squashing everything to this patch, but I see lots of distinct changes:

* IPtables - would be neat if this was a refactor + functional changes in separate patches.
* ctmap changes
* proxymap deletion (We could just unhook the proxymap and leave its code in place, then separately delete the now-unused code)
* The core datapath changes for switching over to tproxy.
* There's maybe one or two other bits but I'm not sure how tightly coupled the changes are across the daemon/datapath

Reply:
iptables refactor was split to a separate PR (merged).
Added a new commit to clean up the remaining references to proxymap.
Old proxymap deletion not done yet.
I'm planning for a follow-up PR to refactor the ctmap lookups. Could also do it now if time allows and merge that before this, but that would be dead code, so maybe not.